aslam_cv_geometric_vision/src/pnp-pose-estimator.cc

```diff
@@ -273,5 +273,132 @@ bool PnpPoseEstimator::absoluteMultiPoseRansac(
   return ransac_success;
 }
 
+bool PnpPoseEstimator::absoluteMultiPoseRansac3DFeatures(
+    const Eigen::Matrix3Xd& measurements,
+    const std::vector<int>& measurement_camera_indices,
+    const Eigen::Matrix3Xd& G_landmark_positions, double ransac_threshold,
+    int max_ransac_iters, aslam::NCamera::ConstPtr ncamera_ptr,
+    aslam::Transformation* T_G_I, std::vector<int>* inliers,
+    std::vector<double>* inlier_distances_to_model, int* num_iters) {
+  CHECK_NOTNULL(T_G_I);
+  CHECK_NOTNULL(inliers);
+  CHECK_NOTNULL(inlier_distances_to_model);
+  CHECK_NOTNULL(num_iters);
+  CHECK_EQ(measurements.cols(), G_landmark_positions.cols()) << "Measurements and landmarks are not in pairs.";
+  if (measurements.size() < 6) {
+    return false;
+  }
+
+  const aslam::Transformation& T_C_B = ncamera_ptr->get_T_C_B(0);
+  aslam::Transformation T_B_C = T_C_B.inverse();
+  Eigen::Matrix3d R_B_C = T_B_C.getRotationMatrix();
+  Eigen::Vector3d p_B_C = T_B_C.getPosition();
+
+  std::vector<Eigen::Vector3d> landmarks;
+  std::vector<Eigen::Vector3d> observations;
+
+  for (int i = 0; i < measurements.cols(); ++i) {
+    landmarks.push_back(G_landmark_positions.col(i));
+    observations.push_back(R_B_C * measurements.col(i) + p_B_C);
+  }
+  Eigen::Matrix3d ransac_rotation_matrix;
+  Eigen::Vector3d ransac_translation;
+  std::vector<size_t> ransac_outliers;
+  std::vector<size_t> ransac_inliers;
+
+  RansacTransformationFor3DPoints(
+      landmarks, observations, &ransac_rotation_matrix, &ransac_translation,
+      &ransac_inliers, &ransac_outliers, ransac_threshold, max_ransac_iters);
+
+  // Set result.
+  T_G_I->getPosition() = ransac_translation;
+  T_G_I->getRotationMatrix() = ransac_rotation_matrix;
+  for (auto inlier : ransac_inliers) {
+    inliers->push_back(inlier);
+    inlier_distances_to_model->push_back(
+        (ransac_rotation_matrix * landmarks[inlier] + ransac_translation -
+         observations[inlier])
+            .norm());
+  }
+  *num_iters = max_ransac_iters;
+
+  return true;
+}
+void PnpPoseEstimator::RansacTransformationFor3DPoints(
+    std::vector<Eigen::Vector3d> point_set_1,
+    std::vector<Eigen::Vector3d> point_set_2,
+    Eigen::Matrix3d* best_rotation_matrix, Eigen::Vector3d* best_translation,
+    std::vector<size_t>* best_inliers, std::vector<size_t>* best_outliers,
+    double ransac_threshold, int ransac_max_iterations) {
+  CHECK_GT(ransac_threshold, 0.0);
+  CHECK_GT(ransac_max_iterations, 0u);
+  CHECK_EQ(point_set_1.size(), point_set_2.size());
+  CHECK_GT(point_set_2.size(), 6u);
+
+  const double ransac_stopping_ration = 0.5;
```

Comment:
Make this a gflag (specifically since the 50% are just a wild guess). BTW shouldn't it be 
```suggestion
  const double pnp_3d_ransac_stopping_ratio = 0.5;
```